lib/DxilPIXPasses/DxilAnnotateWithVirtualRegister.cpp

```diff
@@ -120,12 +143,15 @@ bool DxilAnnotateWithVirtualRegister::runOnModule(llvm::Module &M) {
   }
 
   std::uint32_t InstNum = m_StartInstruction;
-  std::map<llvm::StringRef, std::pair<int, int>> InstructionRangeByFunctionName;
+  std::map<std::string, std::pair<int, int>> InstructionRangeByFunctionName;
```

Comment:
What is the motivation for this? Do you need the map to own the string? If so, `llvm::StringMap` might be a better approach, it is certainly more optimal unless you need ordering.

Reply:
Switched to StringMap